lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,35 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: If a matrix is rows*columns large, than space complexity will be O(rows) or O(columns), depending on which is larger.
 def matrix_convert_to_zero(matrix)
-  raise NotImplementedError
+  rows = matrix.length
+  columns = matrix[0].length
+  
+  sum_rows = []
+  matrix.each do |row|
+    sum_rows.push(row.sum)
+  end
+
+  sum_columns = []
+  columns.times do |i|
+    sum = 0
+    rows.times do |j|
+      sum = sum + matrix[j][i]
+    end
+    sum_columns.push(sum)
+  end
+  
+  max = rows + columns
+
+  rows.times do |i|
+    columns.times do |j|
+      if sum_rows[i] + sum_columns[j] < max
```

Comment:
I have to say, this is really clever.  Nicely done.  It does require you to have two arrays, one for rows and the other for columns.